client/components/domains/use-my-domain/index.jsx

```diff
@@ -101,34 +102,92 @@ function UseMyDomain( {
 		return ! errorMessage;
 	}, [ domainName ] );
 
-	const onNext = useCallback( () => {
+	const setDomainTransferData = useCallback(
+		( isDomainUnlocked ) => {
+			const { LOCKED, UNLOCKED, UNKNOWN } = domainLockStatusType;
+			let lockStatus = UNKNOWN;
+
+			setTransferDomainStepsDefinition(
+				isDomainUnlocked
+					? transferUnlockedDomainStepsDefinition
+					: transferLockedDomainStepsDefinition
+			);
+
+			if ( isDomainUnlocked === null ) {
+				lockStatus = UNKNOWN;
+			} else {
+				lockStatus = isDomainUnlocked ? UNLOCKED : LOCKED;
+			}
+
+			setDomainLockStatus( lockStatus );
+		},
+		[ setTransferDomainStepsDefinition, setDomainLockStatus ]
+	);
+
+	const onNext = useCallback( async () => {
 		if ( ! validateDomainName() ) {
 			return;
 		}
 
 		setIsFetchingAvailability( true );
 		setDomainAvailabilityData( {} );
 
-		wpcom
-			.domain( domainName )
-			.isAvailable( { apiVersion: '1.3', blog_id: selectedSite.ID, is_cart_pre_check: false } )
-			.then( ( availabilityData ) => {
-				const availabilityErrorMessage = getAvailabilityErrorMessage( {
-					availabilityData,
-					domainName,
-					selectedSite,
-				} );
-
-				if ( availabilityErrorMessage ) {
-					setDomainNameValidationError( availabilityErrorMessage );
-				} else {
-					setMode( inputMode.transferOrConnect );
-					setDomainAvailabilityData( availabilityData );
-				}
-			} )
-			.catch( ( error ) => setDomainNameValidationError( error ) )
-			.finally( () => setIsFetchingAvailability( false ) );
-	}, [ domainName, inputMode.transferOrConnect, selectedSite, validateDomainName ] );
+		try {
+			const availabilityData = await wpcom
+				.domain( domainName )
+				.isAvailable( { apiVersion: '1.3', blog_id: selectedSite.ID, is_cart_pre_check: false } );
+
+			let inboundTransferStatusResult = {};
+			try {
+				inboundTransferStatusResult = await wpcom
+					.undocumented()
+					.getInboundTransferStatus( domainName );
+			} catch {
+				availabilityData.status = domainAvailability.TLD_NOT_SUPPORTED;
```

Comment:
```suggestion
				// do nothing
```

I think we can just leave everything as it when this endpoint returns an error since the availability status will already me "mappable" so the transfer option won't be available.

Reply:
Oh, this was old code, sorry. I have removed it already.